assets/aml-benchmark/components/src/aml_benchmark/dataset_preprocessor/dataset_preprocessor.py

```diff
@@ -111,36 +112,54 @@ def prep_using_template(self) -> None:
         template = self.add_json_filter(template)
         env = jinja2.Environment()
         jinja_template = env.from_string(template)
-        with open(self.output_dataset, mode='w', encoding='utf8') as f:
+        with open(self.output_dataset, mode="w", encoding="utf8") as f:
             for example in data:
                 out = jinja_template.render(example)
-                out = out.replace('\'', '\"')
+                out = out.replace("'", '"')
                 out_dict = json.loads(out)
                 if self.encoder_config:
-                    col_to_encode = self.encoder_config.get('column_name')
-                    out_dict[col_to_encode] = self.encoder_config.get(str(out_dict.get(col_to_encode)))
-                logger.info('Loaded dictionary', out_dict)
+                    col_to_encode = self.encoder_config.get("column_name")
+                    out_dict[col_to_encode] = self.encoder_config.get(
+                        str(out_dict.get(col_to_encode))
+                    )
+                logger.info("Loaded dictionary", out_dict)
                 f.write(json.dumps(out_dict) + "\n")
         return
 
     def run(self) -> None:
         """Preprocessor runner."""
+        if self.encoder_config:
+            self.encoder_config = json.loads(self.encoder_config)
         if self.user_preprocessor:
             self.run_user_preprocessor()
             return
-        if self.encoder_config:
-            self.encoder_config = json.loads(self.encoder_config)
         if self.template:
             # all preprocessing will be done based on their provided pattern
             self.prep_using_template()
             return
 
+    def __get_parameters(self):
+        return deepcopy(self.__dict__)
```

Comment:
Could you please help me understand how are we passing these additional arguments? I don't see any modification to the component's spec.yaml as well as to the `DatasetPreprocessor` class.

Thanks!

Reply:
@iamrk04  There is no modification in componen's spec The extra arguments pass is for passing the instance attributes that was set in the components' fields.